src/main/java/sirius/biz/web/BasePageHelper.java

```diff
@@ -192,6 +195,45 @@ public B addEnumFacet(String name, String title, Class<? extends Enum<?>> enumTy
         return addFilterFacet(facet);
     }
 
+    /**
+     * Adds a new facet which is only shown if the given parameter is present in the current {@link WebContext}.
+     * <p>
+     * This can be used to show (and keep) a special parameter in a table view. If the parameter isn't present,
+     * nothing will happen. If a value is given the parameter is shown as facet and it is also kept by using a
+     * facet and an additional hidden facet for its label.
+     *
+     * @param parameterName      the name of the parameter in the web request
+     * @param labelParameterName the name of the parameter which contains the label to show
+     * @param filterField        the name of the database field to filter on
+     * @param title              the label of the facet
+     * @param translator         if no label parameter is present, this translator is used to retrieve one
+     * @return the helper itself for fluent method calls
+     */
+    @SuppressWarnings("unchecked")
+    public B addParameterFacet(@Nonnull String parameterName,
+                               @Nullable String labelParameterName,
+                               @Nonnull Mapping filterField,
+                               @Nonnull String title,
+                               @Nullable ValueComputer<String, String> translator) {
+        Objects.requireNonNull(ctx);
+
+        String value = ctx.get(parameterName).asString();
+
+        if (Strings.isFilled(value)) {
+            Facet facet = new Facet(title, parameterName, value, translator);
+            Value labelAsValue = ctx.get(labelParameterName);
+            facet.addItem(value, labelAsValue.getString(), -1);
```

Comment:
Is this intentional possibly null? Below `asString` is used on the same value to create the second `Facet`. If this is by mistake the Value above could be replaced by a direct call to `asString` or `getString` and `Strings.isFilled` could be used in the `if`

Reply:
if value is null, the parameter is hidden - thats the whole point of this facet. The label can also be null, then the translator will be used (if available). If both are null, the facet is ignored again.